converter/converter.py

```diff
@@ -106,12 +107,25 @@ def convert_data_product_definitions(src: Path, dest: Path) -> bool:
     should_fail_hook = False
     for p in src.glob("**/*.py"):
         spec = importlib.util.spec_from_file_location(name=str(p), location=str(p))
+
+        def styledError(error):
```

Comment:
Function names should in python be written with snake_case.

I'm not a too big fan of helper functions that are defined in the middle of business logic and inside loops. I'd say if we want to define it within the loop, then it should be at the beginning of it, not in the middle of loading the spec. However, I'd prefer to see it being defined outside the loop and preferably even outside the current function, on the main level of the file. By defining it outside the loop, we avoid creating a new function in each iteration of the loop, which is a nice saving if we have a lot of definitions and most of them are OK.

Please also type annotate the parameter(s) it takes (you'll have to add the path when you move it out from the loop).